client/admin-pointers.js

```diff
@@ -22,17 +20,29 @@ jQuery( document ).ready( function( $ ) {
 			return;
 		}
 
+		const target = $( pointer.target );
+
 		const options = $.extend( pointer.options, {
 			close: function() {
 				$.post( ajaxurl, {
 					pointer: pointer.id,
 					action: 'dismiss-wp-pointer',
 				} );
+
+				if ( pointer.dim ) {
+					$( '#wcs-pointer-page-dimmer' ).fadeOut( 500 );
```

Comment:
It would probably be cleaner, but actually `show_pointer()` is only executed for the first valid pointer. We don't have to worry about it executing multiple times.